### ISSUE TITLE
Blead (615e365) Broke on OmniOS under g++

### ISSUE THREAD

**Issue by cjg-cguevara** [Perl/perl5#21058]
Blead (615e365) broke on OmniOS under g++ with the following error:
```
g++ -c -DPERL_CORE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -m64 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -O -Wall -Werror=pointer-arith -Werror=vla -Wextra -Wno-long-long -Wwrite-strings -fPIC pp_sys.c
pp_sys.c: In function 'OP* Perl_pp_syscall()':
pp_sys.c:5815:18: error: 'syscall' was not declared in this scope
 5815 |         retval = syscall(a[0]);
      |                  ^~~~~~~
*** Error code 1
The following command caused the error:
`sh  cflags "optimize='-O'" pp_sys.o` -fPIC pp_sys.c
make: Fatal error: Command failed for target `pp_sys.o'
```

Example:
https://perl5.test-smoke.org/report/5033664
https://perl5.test-smoke.org/file/log_file/5033664

**Comment by jkeenan** [comment]
Blead was buliding and passing on this Solaris variant at d224e8addbbaa4e23cd1690b49c872292646379d (as per https://perl5.test-smoke.org/report/5033631).  Just a few commits later, at https://perl5.test-smoke.org/report/5033664, it was failing to get as far as `miniperl` (as per https://perl5.test-smoke.org/report/5033664).

@Tux, can you take a look?  Thanks.

**Comment by Tux** [comment]
I bet that this will fix it:
```
diff --git a/pp_sys.c b/pp_sys.c
index f101a5a161..b18911b3f4 100644
--- a/pp_sys.c
+++ b/pp_sys.c
@@ -58,6 +58,10 @@
 # endif
 #endif

+#ifdef I_SYS_SYSCALL
+# include <sys/syscall.h>
+#endif
+
 /* XXX Configure test needed.
    h_errno might not be a simple 'int', especially for multi-threaded
    applications, see "extern int errno in perl.h".  Creating such
```

**Comment by tonycoz** [comment]
This should be fixed by 56713ab45217704e3741b391feb10b4ab37c188b.

**Comment by tonycoz** [comment]
Closing since https://github.com/Perl/perl5/commit/56713ab45217704e3741b391feb10b4ab37c188b should have fixed it, and no-one objected.
